packages/flutter_tools/test/commands.shard/hermetic/assemble_test.dart

```diff
@@ -16,20 +16,14 @@ import '../../src/common.dart';
 import '../../src/testbed.dart';
 
 void main() {
-  Testbed testbed;
-  MockBuildSystem mockBuildSystem;
   Cache.disableLocking();
-
-  setUp(() {
-    mockBuildSystem = MockBuildSystem();
-    testbed = Testbed(overrides: <Type, Generator>{
-      BuildSystem: ()  => mockBuildSystem,
-      Cache: () => FakeCache(),
-    });
+  final Testbed testbed = Testbed(overrides: <Type, Generator>{
+    BuildSystem: ()  => MockBuildSystem(),
+    Cache: () => FakeCache(),
   });
 
-  test('Can run a build', () => testbed.run(() async {
-    when(mockBuildSystem.build(any, any, buildSystemConfig: anyNamed('buildSystemConfig')))
+  testbed.test('Can run a build', () async {
```

Comment:
For my own understanding, how is testbed different than using `testUsingContext`?

Reply:
Testbed allows you to supply a setup function which is invoked in the same Zone that the test case is run in. This means that the zone injected values will be exactly the same (logger, fs, buildSystem).

The only way to do this setup with test usingContext is to use top levels initialized in setup and the test case. This requires plumbing values between various setup functions and the test callbacks, which is where nullability issues creep in.

Reply:
Additionally, running it in the same zone allows you to use the same affordances for setup that you would otherwise use for programming. Example:

### before 
```
FileSystem fileSystem;

setUp(() {
  fileSystem = MemoryFileSystem(fileSystemStyle: ...);
  fileSystem.file('a').createSync();
});

testUsingContext('a exists', () {
  expect(fs.file('a).existsSync(), true);
}, overrides: <Type, Generator>{
  FileSystem: () => fileSystem,
}
```

### after
```
Testbed testbed = Testbed(setup: () {
  fs.file('a').createSync();
});

testbed.test('a exists, () {
  expect(fs.file('a).existsSync(), true);
});
```